runtime/tests/tests.rs

```diff
@@ -683,6 +685,147 @@ fn runtime_mock_setup_works() {
 	});
 }
 
+fn apply_extrinsic(origin: AccountId, call: Call) -> Balance {
+	let xt = sign(CheckedExtrinsic {
+		signed: Some((origin, signed_extra(0, 0, None, None))),
+		function: call.clone(),
+	});
+
+	let fee = transfer_fee(&xt, &call);
+
+	Executive::initialize_block(&header());
+	let r = Executive::apply_extrinsic(xt);
+	assert!(r.is_ok());
+
+	fee
+}
+
+#[test]
+fn non_sylo_call_is_not_paid_by_payment_account() {
+	let call = Call::GenericAsset(pallet_generic_asset::Call::transfer(CENTRAPAY_ASSET_ID, dave(), 100));
```

Comment:
suggestion: a single test with a for loop of all the sylo calls would be clearer with less boiler plate

Reply:
I tried that first but then the test needs to be more complicated as there was an issue possibly with applying more than one extrinsic to the same block. On the other hand, we would usually want our test cases to do one test at a time.

Reply:
I guess you'd need to make a new block each loop